hazelcast/src/main/java/com/hazelcast/internal/config/yaml/YamlDomChecker.java

```diff
@@ -34,21 +36,25 @@ private YamlDomChecker() {
     }
 
     /**
-     * Performs {code @null} checks on the provided YAML node recursively.
+     * Performs {@code null} checks on the provided on YAML node recursively.
+     * This check is skipped on the nodes defined as nullable.
      *
      * @param node The YAML node to check for {@code null}s
+     * @param nullableNodes The names of nodes on which null check is not performed
+     *
      */
-    public static void check(YamlNode node) {
+    public static void check(YamlNode node, Set<String> nullableNodes) {
```

Comment:
No strong opinion on this, but could it be `Set<ConfigSections>` instead? You would be able to avoid several `.getName()` calls.

Reply:
We have completely separate config sections classes for clients and members which make this impossible. See: [ClientConfigSections](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/client/config/impl/ClientConfigSections.java), [ConfigSections](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/config/ConfigSections.java).

On the other hand, I would normally remove this `YamlDomChecker` class because the schema validator is already checking for the null values. But, I left it as is because error messages of this may be more preferable than the schema validator messages (+ avoided behavior change).